src/FastSerialization/StreamReaderWriter.cs

```diff
@@ -693,24 +708,33 @@ protected override void Dispose(bool disposing)
                 positionInStream += (uint)position;
                 endPosition = 0;
                 position = 0;
-                // if you are within one read of the end of file, go backward to read the whole block.  
-                uint lastBlock = (uint)(((int)inputStream.Length - bytes.Length + align) & ~(align - 1));
-                if (positionInStream >= lastBlock)
-                {
-                    position = (int)(positionInStream - lastBlock);
-                }
-                else
+                if (inputStream.CanSeek)
                 {
-                    position = (int)positionInStream & (align - 1);
-                }
+                    // if you are within one read of the end of file, go backward to read the whole block.  
+                    uint lastBlock = (uint)(((int)inputStream.Length - bytes.Length + align) & ~(align - 1));
+                    if (positionInStream >= lastBlock)
+                    {
+                        position = (int)(positionInStream - lastBlock);
+                    }
+                    else
+                    {
+                        position = (int)positionInStream & (align - 1);
+                    }
 
-                positionInStream -= (uint)position;
+                    positionInStream -= (uint)position;
+                }
             }
 
             Debug.Assert(positionInStream % align == 0);
             lock (inputStream)
             {
-                inputStream.Seek(positionInStream + endPosition, SeekOrigin.Begin);
+                if(inputStream.CanSeek)
+                {
+                    // for non-seekable streams we always keep the stream positioned at this point
```

Comment:
Is this comment correct?  It talks about non-seekable streams but is in an if block that only applies to seekable streams.

Reply:
the comment was intended to justify why we only do the Seek() in the seekable case but let me update it for better clarity : )